agent/src/agent/pipeline/info.py

```diff
@@ -67,11 +67,19 @@ def _get_timestamp(utc_time):
 def _extract_pipeline_issues(pipeline_info: dict) -> list:
     pipeline_issues = []
     if pipeline_info['issues']['issueCount'] > 0:
-        for i in pipeline_info['issues']['pipelineIssues']:
-            pipeline_issues.append('{severity} - {configGroup} - {configName} - {message}'.format(**i))
+        for issue in pipeline_info['issues']['pipelineIssues']:
+            pipeline_issues.append(_format(issue))
     return pipeline_issues
 
 
+def _format(info: dict) -> str:
+    keys = ['severity', 'configGroup', 'configName', 'message']
+    return ' - '.join(list(filter(
```

Comment:
`' - '.join([info[key] for key in keys if key in info])`
I think this is more understandable than list/filter/map